core/src/test/java/com/linecorp/armeria/internal/server/annotation/AnnotatedDocServiceTest.java

```diff
@@ -450,14 +456,43 @@ void excludeAllServices() throws IOException {
         final AggregatedHttpResponse res = client.get("/excludeAll/specification.json").aggregate().join();
         assertThat(res.status()).isEqualTo(HttpStatus.OK);
         final JsonNode actualJson = mapper.readTree(res.contentUtf8());
-        final JsonNode expectedJson = mapper.valueToTree(new ServiceSpecification(ImmutableList.of(),
-                                                                                  ImmutableList.of(),
-                                                                                  ImmutableList.of(),
-                                                                                  ImmutableList.of(),
-                                                                                  ImmutableList.of()));
+        final Route docServiceRoute = Route.builder().pathPrefix("/excludeAll").build();
+        final ServiceSpecification emptySpecification = new ServiceSpecification(ImmutableList.of(),
+                                                                                 ImmutableList.of(),
+                                                                                 ImmutableList.of(),
+                                                                                 ImmutableList.of(),
+                                                                                 ImmutableList.of(),
+                                                                                 docServiceRoute);
+        final JsonNode expectedJson = mapper.valueToTree(emptySpecification);
         assertThatJson(actualJson).isEqualTo(expectedJson);
     }
 
+    @ParameterizedTest
+    @CsvSource({ "/proxy", "/proxy2/nested" })
+    void rightPathBehindProxy(String proxyPath) throws IOException {
+        server.server().reconfigure(sb -> sb.serviceUnder(proxyPath, (ctx, req) -> {
```

Comment:
I prefer that we don't use `reconfigure` in tests since this can affect tests that might be added afterwards.
(Usually when adding tests to files like this, I think it is more convenient to check the added services in `ServerExtension` instead of having to check each test to see if the server has been configured)

Reply:
Anyway, moved proxies to top-level as pre-defined endpoints for simplicity.

Reply:
If a `server.reconfigure()` is called, the original services defined in `ServerExtension` are not restored.
Thus, if a test is executed after `rightPathBehindProxy` only `/proxy2/nested` and `\docs` service will be available.

Alternatively, we could enable `ServerExtension#runForEachTest`, but it's probably cheaper to just add the service like done in the following commit you added 👍 .